packages/opentelemetry-plugin-grpc/src/grpc.ts

```diff
@@ -0,0 +1,610 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BasePlugin } from '@opentelemetry/core';
+import {
+  Tracer,
+  SpanKind,
+  SpanOptions,
+  Span,
+  Status,
+  CanonicalCode,
+  SpanContext,
+} from '@opentelemetry/types';
+import { GrpcAttributeNames } from './enums/grpcAttributeNames';
+import {
+  grpc,
+  ModuleExportsMapping,
+  ModuleNameToFilePath,
+  GrpcPluginOptions,
+  ServerCallWithMeta,
+  SendUnaryDataCallback,
+  GrpcClientFunc,
+} from './grpc-types';
+import { findIndex } from './grpc-utils';
+
+import * as events from 'events';
+import * as grpcModule from 'grpc';
+import * as shimmer from 'shimmer';
+import * as semver from 'semver';
+import * as path from 'path';
+
+/** The metadata key under which span context is stored as a binary value. */
+export const GRPC_TRACE_KEY = 'grpc-trace-bin';
+
+// tslint:disable-next-line:no-any
+let grpcClientModule: any;
+
+export class GrpcPlugin extends BasePlugin<grpc> {
+  static readonly component = 'grpc';
+
+  options!: GrpcPluginOptions;
+  protected readonly _moduleExports!: grpc;
```

Comment:
Why is this `protected`?

Reply:
Removed these lines in aefc3e4 because they're unneeded